### ISSUE TITLE
Contacts batch API doesn't seem to work for updates (but basic API does)

### ISSUE THREAD

**Issue by wmyers** [HubSpot/hubspot-api-nodejs#548]
Using version `12.0.0`

This works with a call to the basic API, and I can see the Contact properties have been updated:
```
await hubspotClient.crm.contacts.basicApi.update(
     'someone@whoexists.com',
      {
        properties: {
          foo: 'bar'
        },
      },
      'email',
    );
```

This does not fail/return an error, but nothing changes on the Contact:
```
await hubspotClient.crm.contacts.batchApi.update(
  { 
     inputs: [
       {
          id:  'someone@whoexists.com',
          properties: {
            foo: 'bar'
          }
       }
     ]
  }
);
```

**Comment by ksvirkou-hubspot** [comment]
Hi @wmyers 
Need to set `id_property: 'email'`

**Comment by wmyers** [comment]
Hi @ksvirkou-hubspot 
Thanks for your response, but that didn't work. 

`id_property` is not recognised as a property I think. I tried with `idProperty` and that did throw an error: `Unable to perform update/upsert by non-unique 0-1 property`.

This lead to [this thread](https://community.hubspot.com/t5/APIs-Integrations/Need-help-with-the-batch-update-API-quot-Unable-to-perform/td-p/927706) and finally to [these docs](https://developers.hubspot.com/docs/api/crm/contacts#upsert-contacts) where it seems to say (not enormously clearly) that the `update` endpoint only works for batches if you use the record ID values for the `id`, but further down it says the `upsert` endpoint does work for batches with the email for the `id`.

So I ran:
```
await hubspotClient.crm.contacts.batchApi.upsert(
  { 
     inputs: [
       {
          id:  'someone@whoexists.com',
          idProperty: 'email',
          properties: {
            foo: 'bar'
          }
       }
     ]
  }
);
```
and this worked.
